cpp/src/dictionary/add_keys.cu

```diff
@@ -72,15 +72,16 @@ std::unique_ptr<column> add_keys( dictionary_column_view const& dictionary_colum
                     mr, stream);
     // now create the indices column -- map old values to the new ones
     // gather([4,0,3,1,2,2,2,4,0],[0,1,2,3,5]) = [5,0,3,1,2,2,2,5,0]
-    column_view indices_view( data_type{INT32}, dictionary_column.size(), 
+    column_view indices_view( data_type{INT32}, dictionary_column.size(),
                               dictionary_column.indices().data<int32_t>(),
                               nullptr, 0, dictionary_column.offset() );
     auto table_indices = cudf::experimental::detail::gather( table_view{{map_indices->view()}},
-                                                             indices_view, false, false, false,
+                                                             indices_view, false, true, false, // ignore out-of-bounds
```

Comment:
When will the indices be out of bounds?

Reply:
If the dictionary column contains nulls then the corresponding indices values are technically invalid and could be any value including those that are out of bounds of the keys column.

Reply:
Ah, right, makes sense. So in the case where an index value is null, but within bounds, we'll be gathering a random value from the `keys`? Is there any potentially negative consequence of that?

Reply:
Not at all. At this point we are just gather integers and do not need to reference any of the keys. Or do you mean in general? Just like any other column, reading the row value for a null-entry would be undefined -- no one should dereference into the keys column with an index value for a null row entry.